src/visualisation/trigram.cc

```diff
@@ -119,77 +119,98 @@ void TrigramWidget::refresh(AdditionalResampleDataPtr ad) {
   doneCurrent();
 }
 
-bool TrigramWidget::prepareOptionsPanel(QBoxLayout *layout) {
-  VisualisationWidget::prepareOptionsPanel(layout);
+void TrigramWidget::prepareOptions(QMainWindow *visualisation_window) {
+  VisualisationWidget::prepareOptions(visualisation_window);
 
-  QLabel *brightness_label = new QLabel("Brightness: ");
-  brightness_label->setAlignment(Qt::AlignTop);
+  QColor icon_color = palette().color(QPalette::WindowText);
+  visualisation_window->addToolBarBreak();
+
+  /////////////////////////////////////
+  // Camera manipulators
+  prepareManipulatorToolbar(visualisation_window);
+
+  /////////////////////////////////////
+  // Shape
+  QToolBar* shape_toolbar = new QToolBar("Shape", this);
+  shape_toolbar->setMovable(false);
+  cube_action_ = new QAction(this);
```

Comment:
addAction doesn't take ownership, but actions have their parents (QObject) that handle deletion process.